### PR TITLE
Course details API call and Unit tests

### PR DESCRIPTION
**Pull request by snipered2003**

### CODE REVIEW

File: DFC.App.FindACourse/Controllers/DetailsController.cs [hunk 1]
```diff
@@ -3,26 +3,28 @@
...
         [Route("find-a-course/course/body/course/details")]
-        public IActionResult Details(string courseId, string runId)
+        public async Task<IActionResult> Details(string courseId, string runId, string searchTerm)
         {
             this.logger.LogInformation($"{nameof(this.Details)} has been called");
             var model = new DetailsViewModel();
+            model.SearchTerm = searchTerm;
 
             if (string.IsNullOrEmpty(courseId) || string.IsNullOrEmpty(runId))
             {
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse/Controllers/DetailsController.cs]: When I looked at this file its throwing a generic exception, can you check that please think in need to be a more specific one.

File: DFC.App.FindACourse/Controllers/DetailsController.cs [hunk 2]
```diff
@@ -31,7 +33,7 @@ public IActionResult Details(string courseId, string runId)
 
             try
             {
-               model.courseDetails = this.findACourseService.GetCourseDetails(courseId, runId).Result;
+               model.CourseDetails = this.findACourseService.GetCourseDetails(courseId, runId).Result;
             }
             catch (Exception ex)
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse/Controllers/DetailsController.cs]: Also when you this exception its swallowing it and logging it as information.

File: DFC.App.FindACourse/Controllers/DetailsController.cs (hunk 1 above)
Comment (snipered2003) [reply DFC.App.FindACourse/Controllers/DetailsController.cs]: Changed

File: DFC.App.FindACourse/Controllers/DetailsController.cs (hunk 2 above)
Comment (snipered2003) [reply DFC.App.FindACourse/Controllers/DetailsController.cs]: I've changed this to log error. Good spot.

**Pull request merged**